enginetest/index_queries.go

```diff
@@ -442,9 +442,11 @@ var ComplexIndexQueries = []ScriptTest{
 				Query:    "SELECT * FROM test WHERE ((v1>=23) OR (v1=47 AND v2>23));",
 				Expected: []sql.Row{{58, 56, 0}, {61, 57, 49}, {72, 65, 80}, {85, 81, 4}, {49, 45, 86}, {50, 46, 46}, {62, 58, 12}, {92, 86, 88}, {47, 45, 31}, {54, 50, 0}, {55, 50, 14}, {87, 83, 30}, {91, 86, 56}, {66, 59, 54}, {76, 69, 34}, {79, 76, 39}, {46, 45, 22}, {57, 54, 38}, {68, 61, 3}, {93, 87, 51}, {45, 44, 67}, {52, 48, 22}, {30, 28, 83}, {53, 49, 0}, {69, 61, 34}, {73, 65, 97}, {90, 84, 45}, {82, 79, 36}, {41, 42, 0}, {43, 43, 63}, {65, 59, 45}, {100, 98, 61}, {95, 93, 19}, {56, 51, 35}, {59, 56, 60}, {67, 60, 66}, {77, 72, 52}, {89, 84, 9}, {24, 24, 60}, {33, 34, 22}, {35, 35, 89}, {63, 58, 32}, {83, 80, 61}, {39, 39, 86}, {26, 25, 31}, {29, 28, 24}, {38, 39, 55}, {40, 40, 97}, {74, 67, 95}, {78, 74, 81}, {81, 78, 90}, {88, 83, 74}, {28, 27, 24}, {37, 38, 66}, {48, 45, 63}, {51, 47, 5}, {64, 59, 29}, {80, 78, 0}, {86, 82, 16}, {96, 93, 21}, {98, 98, 0}, {25, 25, 14}, {27, 27, 9}, {32, 33, 39}, {75, 68, 11}, {84, 80, 88}, {99, 98, 51}, {31, 31, 14}, {44, 44, 48}, {60, 57, 29}, {70, 63, 19}, {71, 63, 69}, {34, 34, 91}, {36, 38, 20}, {42, 42, 82}, {94, 89, 3}, {97, 93, 96}},
 			}, {
+				//TODO why don't we eliminate filter with index scan?
 				Query:    "EXPLAIN SELECT * FROM test WHERE ((v1=30) OR (v1<>67));",
 				Expected: []sql.Row{{"Filter((test.v1 = 30) OR (NOT((test.v1 = 67))))"}, {" └─ Projected table access on [pk v1 v2]"}, {"     └─ IndexedTableAccess(test on [test.v1,test.v2])"}},
 			}, {
+				//TODO why don't we eliminate filter with index scan?
```

Comment:
Probably because of how you're reconstructing the expression in indexes.go. Points to a possible bug.

Reply:
The problem is that expressions are not isomorphic across the filter boundary. So nots and inversions are different expressions that I can't easily compare right now.